src/main/java/core/basesyntax/DataConverterImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+        for (String line : data) {
+            if (line.startsWith("type")) {
+                continue;
+            }
+            String[] parts = line.split(",");
+            FruitTransaction.Operation operation = getOperationByCode(parts[0]);
+            String fruit = parts[1];
```

Comment:
Make sure to validate the parts of the data to avoid `ArrayIndexOutOfBoundsException` if the input data is malformed.